src/compiler/diagnosticMessages.json

```diff
@@ -2726,6 +2726,14 @@
         "category": "Error",
         "code": 2774
     },
+    "Control flow effects of calls to assertion and never-returning functions are reflected only when every variable or property referenced in the function expression is declared with an explicit type annotation.": {

```

Comment:
I don't know what this is telling a user. Is it something like the following?

```
This assertion call cannot be properly analyzed because not every variable and property used in the current function has an explicit type annotation.

This 'never'-returning function call cannot be properly analyzed because not every variable and property used in the current function has an explicit type annotation.
```

Along with

```
This assertion call cannot be properly analyzed because the function being called is not a qualified name.

This 'never'-returning function call cannot be properly analyzed because the function being called is not a qualified name.
```

Reply:
1. I think that "explicitly never-returning function" is really just another kind of assertion. We could introduce the concept of *guard* = "explicitly never-returning function" if it's too ambiguous. 
2. Mentioning control flow isn't necessary for fixing the error -- and presumably most people understand the concept of assertion better than control flow anyway.
3. As for @DanielRosenwasser's wording: "This assertion call" is implied by the error span. The fact that it's an assertion is implied by the error span+error wording. So neither of those is necessary. Wording this categorically also allows us to flip the message back to positive.

```
Assertions must be an identifier or qualified name.
```


```
Assertions must reference a declaration with an explicit type annotation.
  Related: This is the declaration that should be annotated.
```

4. We know that the *expression* has type annotations because `hasTypePredicateOrNeverReturnType(signature)` is true.
5. If we want to handle the case where there is more than one declaration, the message can be

```
Assertions must reference declaration(s) with an explicit type annotation.
  Related: This is one declaration that must be annotated.
  Related: This is one declaration that must be annotated.
  :  
  : 
```